src/woo_publications/publications/admin.py

```diff
@@ -156,6 +158,60 @@ def reassess_retention_policy(
     )
 
 
+@admin.action(description=_("revoke the selected %(verbose_name_plural)s"))
+def revoke_instance(
+    modeladmin: PublicationAdmin | DocumentAdmin | TopicAdmin,
+    request: HttpRequest,
+    queryset: models.QuerySet[Publication | Document | Topic],
+):
+    assert is_authenticated_request(request)
+    model = queryset.model
+    num_objects = queryset.count()
+
+    if model is Publication:
+        task_fn = remove_publication_from_index
+        kwarg_name = "publication_id"
+    elif model is Document:
+        task_fn = remove_document_from_index
+        kwarg_name = "document_id"
+    elif model is Topic:
+        task_fn = remove_topic_from_index
+        kwarg_name = "topic_id"
+    else:  # pragma: no cover
+        raise ValueError("Unsupported model: %r", model)
+
+    for obj in queryset.iterator():
```

Comment:
should be apply an additional filter that excludes (already) revoked instances? Possibly only filter on "published" records.

Reply:
I think that wanting to revoke concepts is a valid thing to do, so I made sure to ignore all the revoked items.